### DISCUSSION TITLE
Socket io {_placeholder:true,num:0}

### DISCUSSION THREAD

**Discussion by ughurabbasov** [socketio/socket.io#4927] [Q&A]
<img width="857" alt="Screenshot 1445-07-08 at 21 59 30" src="https://github.com/socketio/socket.io/assets/89932220/f312d85b-216e-49ee-b473-1a91d129cf16">
As we see there shows how socket io sending data in binary format. Is there possible way to send this Buffer without placeholder. I want to send everything only with binary message without additional placeholder. If you are going to say socket io parser then is it going to solve this. because i already tried another parsers which couldn't solve it

**Comment by darrachequesne** [comment]
A custom parser like the one based on [msgpack](https://github.com/socketio/socket.io-msgpack-parser) should work.

The packet and its binary attachments are all concatenated in a single packet.

Documentation: https://socket.io/docs/v4/custom-parser/
Example: https://github.com/socketio/socket.io/tree/main/examples/custom-parsers
